java/src/main/java/org/apache/arrow/gandiva/evaluator/Filter.java

```
@@ -0,0 +1,145 @@
+   * @return A native filter object that can be used to invoke on a RecordBatch
+   */
+  public static Filter make(Schema schema, Condition condition) throws GandivaException {
+    return make(schema, condition, ConfigurationBuilder.getDefaultConfiguration());
+  }
+
+  /**
+   * Invoke this function to generate LLVM code to evaluate the condition expression.
+   * Invoke Filter::Evaluate() against a RecordBatch to evaluate the filter on
+   * this record batch
+   *
+   * @param schema          Table schema. The field names in the schema should match the fields used
+   *                        to create the TreeNodes
+   * @param condition       condition to be evaluated against data
+   * @param configurationId Custom configuration created through config builder.
+   * @return A native evaluator object that can be used to invoke these projections on a RecordBatch
+   */
+  public static Filter make(Schema schema, Condition condition, long configurationId)
+      throws GandivaException {
+    // Invoke the JNI layer to create the LLVM module representing the filter.
+    GandivaTypes.Condition conditionBuf = condition.toProtobuf();
+    GandivaTypes.Schema schemaBuf = ArrowTypeHelper.arrowSchemaToProtobuf(schema);
+    JniWrapper gandivaBridge = JniWrapper.getInstance();
+    long moduleId = gandivaBridge.buildFilter(schemaBuf.toByteArray(),
+        conditionBuf.toByteArray(), configurationId);
+    return new Filter(moduleId, schema);
+  }
+
+  /**
+   * Invoke this function to evaluate a set of expressions against a recordBatch.
+   *
+   * @param recordBatch Record batch including the data
+   * @param selectionVector  Result of applying the filter on the data
+   */
+  public void evaluate(ArrowRecordBatch recordBatch, SelectionVector selectionVector)
+      throws GandivaException {
+
+    if (this.closed) {
+      throw new EvaluatorClosedException();
+    }
+
+    int numRows = recordBatch.getLength();
```

should we also cross validate the record batch schema with make schema here to fail fast..this is not present in projector too..maybe add in both?
(looks like we do that in projector and filter in gandiva core..)

---

evaluate codepath is very performance sensitive. can't afford to do the same check twice.